scripts/utils.py

```diff
@@ -38,6 +39,10 @@ def get_dict_difference(parent_dict, child_dict):
 
 
 def combine_dicts(primary_dict, supplementary_dict):
+    if not primary_dict:
+        return supplementary_dict
+    elif not supplementary_dict:
+        return primary_dict
```

Comment:
```suggestion
    if not primary_dict or not supplementary_dict:
        return primary_dict or supplementary_dict
```

this is simpler, but I'm not sure about the readability of my code :thinking: